staging/src/k8s.io/apiserver/pkg/endpoints/filters/authentication.go

```diff
@@ -98,6 +101,9 @@ func WithAuthentication(handler http.Handler, auth authenticator.Request, failed
 		if len(apiAuds) > 0 {
 			req = req.WithContext(authenticator.WithAudiences(req.Context(), apiAuds))
 		}
+		ctx := req.Context()
+		ae := request.AuditEventFrom(ctx)
+		audit.LogAnnotation(ae, annotationKey, getAuthMethods(req))
```

Comment:
This method is just for providing a failure message, it doesn't actually give the method used.

Reply:
I am looking at 
```
func (w *WebhookTokenAuthenticator) AuthenticateToken(ctx context.Context, token string) (*authenticator.Response, bool, error) {
```
to see if useful information for audit can be obtained.